src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -114,16 +122,21 @@ public boolean requiresReference(){
 
     @Override
     protected void runTool(final JavaSparkContext ctx) {
-        final List<SimpleInterval> intervals = hasIntervals() ? getIntervals() : IntervalUtils.getAllIntervalsForReference(getHeaderForReads().getSequenceDictionary());
-        final JavaRDD<VariantContext> variants = callVariantsWithHaplotypeCaller(getAuthHolder(), ctx, getReads(), getHeaderForReads(), getReference(), intervals, hcArgs, shardingArgs);
+        // Reads must be coordinate sorted to use the overlaps partitioner
+        final SAMFileHeader readsHeader = getHeaderForReads().clone();
+        readsHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
```

Comment:
I think we're going to want an option to override this at some point for performance reasons, but we can do that as a later upgrade.